test/simple/test-use-strict-builtin-modules.js

```diff
@@ -0,0 +1,53 @@
+// Copyright Joyent, Inc. and other Node contributors.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to permit
+// persons to whom the Software is furnished to do so, subject to the
+// following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+/*
+ * This test makes sure that every builtin module can be loaded
+ * when the V8's --use-strict command line option is passed to node.
+ */
+
+var child_process = require('child_process');
+var assert = require('assert');
+
+function shouldBeTested(builtinModuleName) {
+  assert(typeof builtinModuleName === 'string',
+         'builtinModuleName must be a string');
+  assert(builtinModuleName.length > 0,
+         'builtinModuleName must be a non-empty string');
+
+  if (builtinModuleName === 'module') {
+    // Do not load 'module' because:
+    // 1) It's loaded as part of the normal node.js startup process.
+    // 2) Loading it twice is not supported since it requires
+    // that NativeModule.require is the existing require implementation,
+    // but it overrides it when it's loaded as part of node's bootstrap.
+    return false;
+  }
+
+  return true;
+}
+
+if (process.argv[2] !== 'child') {
+  child_process.fork(__filename, ['child'], { execArgv: ['--use-strict']});
+} else {
+  var natives = process.binding('natives');
+  Object.keys(natives).filter(shouldBeTested).forEach(require);
```

Comment:
Is filtering really necessary here? I ran this locally on 0.12 without the filter, and it worked fine.